api/service_registrations.go

```diff
@@ -0,0 +1,129 @@
+package api
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// ServiceRegistrations is used to query the service endpoints.
+type ServiceRegistrations struct {
+	client *Client
+}
+
+// ServiceRegistration is the fully hydrated object which contains all details
+// of an individual service registration.
+type ServiceRegistration struct {
+
+	// ID is the unique identifier for this registration. It currently follows
+	// the Consul service registration format to provide consistency between
+	// the two solutions.
+	ID string
+
+	// ServiceName is the human friendly identifier for this service
+	// registration.
+	ServiceName string
+
+	// Namespace represents the namespace within which this service is
+	// registered.
+	Namespace string
+
+	// NodeID is Node.ID on which this service registration is currently
+	// running.
+	NodeID string
+
+	// Datacenter is the DC identifier of the node as identified by
+	// Node.Datacenter.
+	Datacenter string
+
+	// JobID is Job.ID and represents the job which contained the service block
+	// which resulted in this service registration.
+	JobID string
+
+	// AllocID is Allocation.ID and represents the allocation within which this
+	// service is running.
+	AllocID string
+
+	// Tags are determined from either Service.Tags or Service.CanaryTags and
+	// help identify this service. Tags can also be used to perform lookups of
+	// services depending on their state and role.
+	Tags []string
+
+	// Address is the IP address of this service registration. This information
+	// comes from the client and is not guaranteed to be routable; this depends
+	// on cluster network topology.
+	Address string
+
+	// Port is the port number on which this service registration is bound. It
+	// is determined by a combination of factors on the client.
+	Port int
+
+	CreateIndex uint64
+	ModifyIndex uint64
+}
+
+// ServiceRegistrationListStub is the list stub response when performing a
+// listing of all service registrations within a namespace. The stub response
+// is used to ensure performance within clusters with a large number of
+// registrations; particularly when used with blocking queries.
+type ServiceRegistrationListStub struct {
+
+	// Services is an array of service registration stubs, grouped by their
+	// namespace. The length should always be one, unless the query is made
+	// using the namespace wildcard parameter.
+	Services []*ServiceRegistrationStub
+
+	CreateIndex uint64
+	ModifyIndex uint64
```

Comment:
Aren't these values going to be set in the RPC handler on the `QueryMeta` already? It looks like that's being handled in the HTTP endpoint too.

(And if we didn't have these fields, we could avoid having a separate struct for ListStub instead of sending back a slice of stubs, I think?)

Reply:
I think you're completely right, thanks! I'll make this change now.